src/Akka.Streams.Kafka/Stages/Producers/DefaultProducerStage.cs

```diff
@@ -79,12 +73,23 @@ public DefaultProducerStageLogic(IProducerStage<K, V, P, TIn, TOut> stage, Attri
                         {
                             var result = new TaskCompletionSource<IResults<K, V, P>>();
                             AwaitingConfirmation.IncrementAndGet();
-                            Producer.Produce(message.Record, BuildSendCallback(result, onSuccess: report =>
+                            try
                             {
-                                result.SetResult(new Result<K, V, P>(report, message));
-                            }));
-                            PostSend(msg);
-                            Push(stage.Out, result.Task as Task<TOut>);
+                                var callback = BuildSendCallback(
+                                    completion: result, 
+                                    onSuccess: report =>
+                                    {
+                                        result.SetResult(new Result<K, V, P>(report, message));
+                                    }, 
+                                    onFailure: OnProduceFailure);
+                                Producer.Produce(message.Record, GetAsyncCallback(callback));
+                                PostSend(msg);
+                                Push(stage.Out, result.Task as Task<TOut>);
+                            }
+                            catch (Exception exception)
+                            {
+                                OnProduceFailure(exception);
```

Comment:
Enclose the Producer.Produce with a try..catch and route any exceptions to the OnProduceFailure method.